File: asyncjs/pokemon.js
// introduced in 2015 with ES6

// fetch("https://pokeapi.co/api/v2/pokemon/ditto")
//     .then((res) => res.json())
//     .then((json) => {
//         console.log(json);
//     });

// introduced in 2017 with ES2017

async function getPokemon() {
    let sumOfWeights = 0;
    const numberOfPokemonGen1 = 151;

    for (let i = 0; i < numberOfPokemonGen1; i++) {
        const response = await fetch(
            `https://pokeapi.co/api/v2/pokemon/${i + 1}`
        );
        const data = await response.json();
        sumOfWeights += data.weight;
        console.log(`currently processing ${data.name}...`);
    }

    console.log(sumOfWeights / numberOfPokemonGen1);
}

getPokemon();

File: notes/Python/Intermediate/week7/pokemon.py
import requests

url = "https://pokeapi.co/api/v2/pokemon/"

# first loop will get average weight
total_weight = 0
for i in range(151):
  response = requests.get(url + str(i+1))
  data = response.json()
  total_weight += data["weight"]
  print("getting weight for: " + data["name"])

average_weight = total_weight / 151
print(average_weight)

above_average_weight = 0
for i in range(151):
  response = requests.get(url + str(i+1))
  data = response.json()
  if (data["weight"] > average_weight):
    print(data["name"] + " is over the average weight!")
    above_average_weight += 1

print(above_average_weight)